src/Microsoft.DocAsCode.Build.TableOfContents/TocResolver.cs

```diff
@@ -59,56 +107,80 @@ private TocItemInfo ResolveItemCore(TocItemInfo wrapper, Stack<FileAndType> stac
                         }
-                        if (string.IsNullOrEmpty(item.Homepage) && string.IsNullOrEmpty(item.HomepageUid))
+                        if (string.IsNullOrEmpty(item.TopicHref) && string.IsNullOrEmpty(item.TopicUid))
                         {
                             var defaultItem = GetDefaultHomepageItem(item);
                             if (defaultItem != null)
                             {
-                                item.Homepage = defaultItem.Href;
-                                item.HomepageUid = defaultItem.Uid;
+                                item.TopicHref = defaultItem.Href;
+                                item.TopicUid = defaultItem.TopicUid;
                             }
                         }
                     }
+                    // for backward compatibility
+                    item.Homepage = item.TopicHref;
+                    if (item.Href == null)
+                    {
+                        item.Href = item.TocHref;
+                    }
 
                     break;
                 case HrefType.RelativeFolder:
                     {
-                        var relativeFolder = (RelativePath)file.File + (RelativePath)item.Href;
-                        var tocFilePath = relativeFolder + (RelativePath)Constants.YamlTocFileName;
-
-                        var tocFile = new FileAndType(file.BaseDir, tocFilePath, file.Type, file.PathRewriter);
                         TocItemInfo tocFileModel;
-
-                        // First, try finding toc.yml under the relative folder
-                        // Second, try finding toc.md under the relative folder
-                        if (!_collection.TryGetValue(tocFile, out tocFileModel))
+                        if (!string.IsNullOrEmpty(item.TocHref) && (tocHrefType == HrefType.MarkdownTocFile || tocHrefType == HrefType.YamlTocFile))
                         {
-                            tocFilePath = relativeFolder + (RelativePath)Constants.MarkdownTocFileName;
-                            tocFile = new FileAndType(file.BaseDir, tocFilePath, file.Type, file.PathRewriter);
+                            Logger.LogWarning($"Href {item.Href} is overwritten by tocHref {item.TocHref}");
+                            var tocFilePath = (RelativePath) file.File + (RelativePath) item.TocHref;
+
+                            var tocFile = new FileAndType(file.BaseDir, tocFilePath, file.Type, file.PathRewriter);
+
                             if (!_collection.TryGetValue(tocFile, out tocFileModel))
                             {
-                                var message = $"Unable to find either {Constants.YamlTocFileName} or {Constants.MarkdownTocFileName} inside {item.Href}. Make sure the file is included in config file docfx.json!";
+                                var message = 
+                                    $"Unable to find either {item.TocHref}. Make sure the file is included in config file docfx.json!";
```

Comment:
we should also do this check if href is null

---
In reply to: [73300585](https://github.com/dotnet/docfx/pull/537#discussion_r73300585) [](ancestors = 73300585)